### ISSUE TITLE
[C#] Implement new Concept Exercise: exceptions

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#966]
This issue describes how to implement the `exceptions` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the Concept of Exceptions in C#.

## Learning objectives

- Know what exceptions are.
- Know when an exception should be thrown.
- Know how to throw (and rethrow) an exception.
- Know how to catch an exception.
- Know the most important built-in exceptions (`ArgumentException`, `InvalidOperationException`).

## Out of scope

- Exception filters.
- User-defined exceptions.
- try/finally blocks
- Memory and performance characteristics.

## Concepts

This Concepts Exercise's Concepts are:

- `exceptions`: know what exceptions are; know when an exception should be thrown; know how to throw (and rethrow) an exception; know how to catch an exception; know the most important built-in exceptions (`ArgumentException`, `InvalidOperationException`).

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `classes`: defining a class for a custom exception and using methods.

### Hints

- [Exceptions][exceptions]: basic introduction to exceptions.
- [Using exceptions][using-exceptions]: how to use exceptions.
- [Exception handling][exception-handling]: how to catch exceptions.
- [Creating and throwing exceptions][creating-and-throwing-exceptions]: how to create and throw exceptions.

### After

- [Exceptions][exceptions]: basic introduction to exceptions.
- [Using exceptions][using-exceptions]: how to use exceptions.
- [Exception handling][exception-handling]: how to catch exceptions.
- [Creating and throwing exceptions][creating-and-throwing-exceptions]: how to create and throw exceptions.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/types/tuple.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[exceptions]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/
[using-exceptions]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/using-exceptions
[exception-handling]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/exception-handling
[creating-and-throwing-exceptions]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/creating-and-throwing-exceptions

**Comment by archrisV** [comment]
Hi, I would like to try and tackle this issue.

**Comment by ErikSchierboom** [comment]
@archrisV Great!

**Comment by ErikSchierboom** [comment]
@archrisV I've updated this issue, as we're re-assessing the scope of the various exercises. Practically, what this means for this exercise is:

- User-defined (custom) exceptions are out of scope.
- Exception filtering is out of scope.
